checks/allprocesses_windows.go

```
@@ -0,0 +1,306 @@
+		PrivatePageCount      uint64
+		TerminationDate       *time.Time
+		UserModeTime          uint64
+		WorkingSetSize        uint64
+	*/
+}
+
+type IO_COUNTERS struct {
+	ReadOperationCount  uint64
+	WriteOperationCount uint64
+	OtherOperationCount uint64
+	ReadTransferCount   uint64
+	WriteTransferCount  uint64
+	OtherTransferCount  uint64
+}
+
+func getProcessMemoryInfo(h syscall.Handle, mem *process.PROCESS_MEMORY_COUNTERS) (err error) {
+	r1, _, e1 := syscall.Syscall(procGetProcessMemoryInfo.Addr(), 3, uintptr(h), uintptr(unsafe.Pointer(mem)), uintptr(unsafe.Sizeof(*mem)))
+	if r1 == 0 {
+		if e1 != 0 {
+			err = error(e1)
+		} else {
+			err = syscall.EINVAL
+		}
+	}
+	return
+}
+func getProcessHandleCount(h syscall.Handle, count *uint32) (err error) {
+	r1, _, e1 := procGetProcessHandleCount.Call(uintptr(h), uintptr(unsafe.Pointer(count)))
+	if r1 == 0 {
+		return e1
+	}
+	return nil
+}
+
+func getProcessIoCounters(h syscall.Handle, counters *IO_COUNTERS) (err error) {
+	r1, _, e1 := procGetProcessIoCounters.Call(uintptr(h), uintptr(unsafe.Pointer(counters)))
+	if r1 == 0 {
+		return e1
+	}
+	return nil
+}
+
+func getProcessMapFromWmi() (map[uint32]Win32_Process, error) {
+	var dst []Win32_Process
+	q := wmi.CreateQuery(&dst, "")
+	err := wmi.Query(q, &dst)
+	if err != nil {
+		return map[uint32]Win32_Process{}, err
+	}
+	if len(dst) == 0 {
+		return map[uint32]Win32_Process{}, fmt.Errorf("could not get Process")
+	}
+	results := make(map[uint32]Win32_Process)
+	for _, proc := range dst {
+		results[proc.ProcessID] = proc
+	}
+	return results, nil
+
+}
+func getAllProcesses() (map[int32]*process.FilledProcess, error) {
+	allProcsSnap := w32.CreateToolhelp32Snapshot(w32.TH32CS_SNAPPROCESS, 0)
+	if allProcsSnap == 0 {
+		return nil, syscall.GetLastError()
+	}
+	procs := make(map[int32]*process.FilledProcess)
+
+	defer w32.CloseHandle(allProcsSnap)
+	var pe32 w32.PROCESSENTRY32
+	pe32.DwSize = uint32(unsafe.Sizeof(pe32))
+
+	fromWmi, _ := getProcessMapFromWmi()
```

why is this error getting ignored?